.github/workflows/build-hg.yml

```diff
@@ -0,0 +1,78 @@
+name: Build Mercurial
+run-name: Build Mercurial ${{ inputs.hg-version }}
+
+on:
+  # Run manually
+  workflow_dispatch:
+    inputs:
+      hg-version:
+        description: 'Mercurial version to build'
+        required: false
+        type: string
+        default: '6.5.1'
+
+jobs:
+  build-python:
+    # Use specific Ubuntu version rather than ubuntu-latest, so we can control when to upgrade the runner
+    runs-on: ubuntu-22.04
+    strategy:
+      matrix:
+        python-version:
+          - "3.8"
+          - "3.9"
+          - "3.10"
+          - "3.11"
+          - "3.12"
+          - "3.13"
+    steps:
+    - name: Setup Python
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+        allow-prereleases: true
+    - name: Install Python setuptools
+      run: pip install setuptools
+    - name: Checkout Mercurial
+      run: hg clone https://repo.mercurial-scm.org/hg
+    - name: Checkout hg version to build
+      run: hg checkout "${VER}"
+      working-directory: hg
+      env:
+        VER: ${{ inputs.hg-version }}
+    - name: Build Mercurial
+      run: make local
+      working-directory: hg
+    - name: Delete .hg folder prior to upload
+      run: rm -rf .hg
+      working-directory: hg
+    - name: Upload Mercurial build results
+      uses: actions/upload-artifact@v4
+      with:
+        name: hg-for-python-${{ matrix.python-version }}
+        path: |
+          hg/contrib
+          hg/COPYING
+          hg/hg
+          hg/hgdemandimport
+          hg/hgext
+          hg/mercurial
+        if-no-files-found: error
+
+  combine:
+    runs-on: ubuntu-latest
+    needs: build-python
+    steps:
+
+    - name: Combine Mercurial builds
```

Comment:
I don't get this 🤔. If we merge them all together, what is the result? Who wins conflicts?

Reply:
Each step in the matrix uses a different Python version. The way Python handles compiled C extensions, it only guarantees ABI stability for a specific Python version (major.minor), and the version number is embedded in the filename, e.g. `zstd.cpython-312-x86_64-linux-gnu.so` for Python 3.12. By merging all the builds together, the last one alphabetically wins conflicts when files have the same name, but since the files with the same name are .py files with no changes between the different versions in the build matrix, it doesn't matter which one wins.

But the compiled C extensions have different filenames because they were compiled with different versions of Python, so you end up with the following files in the distributed package:

```
zstd.cpython-38-x86_64-linux-gnu.so
zstd.cpython-39-x86_64-linux-gnu.so
zstd.cpython-310-x86_64-linux-gnu.so
zstd.cpython-311-x86_64-linux-gnu.so
zstd.cpython-312-x86_64-linux-gnu.so
zstd.cpython-313-x86_64-linux-gnu.so
```

The result is that no matter which version of Linux these files end up on, `import zstd` will work. On Ubuntu 22.04, which has Python 3.10, `import zstd` will load `zstd.cpython-310-x86_64-linux-gnu.so`. On Ubuntu 24.04, which has Python 3.12, `import zstd` will load `zstd.cpython-312-x86_64-linux-gnu.so`. (Ubuntu 24.04 wasn't yet released when I set this up, but because I did this merge step, we don't need a new version of the NuGet package in order to support 24.04).